pkg/config/organization.go

```diff
@@ -0,0 +1,175 @@
+	orgGroup, ok := config.ChannelGroup.Groups[OrdererGroupKey].Groups[orgName]
+	if !ok {
+		return Organization{}, fmt.Errorf("orderer org %s does not exist in channel config", orgName)
+	}
+
+	org, err := getOrganization(orgGroup, orgName)
+	if err != nil {
+		return Organization{}, err
+	}
+
+	// Orderer organization requires orderer endpoints.
+	endPointsProtos := &cb.OrdererAddresses{}
+	err = unmarshalConfigValueAtKey(orgGroup, EndpointsKey, endPointsProtos)
+	if err != nil {
+		return Organization{}, err
+	}
+	ordererEndpoints := []string{}
+	for _, address := range endPointsProtos.Addresses {
+		ordererEndpoints = append(ordererEndpoints, address)
+	}
+	org.OrdererEndpoints = ordererEndpoints
+
+	// Remove AnchorPeers which are application org specific.
+	org.AnchorPeers = nil
+
+	return org, err
+}
+
+// GetConsortiumOrg retrieves a existing org from a consortium organization config group.
+func GetConsortiumOrg(config cb.Config, consortiumName, orgName string) (Organization, error) {
+	consortium, ok := config.ChannelGroup.Groups[ConsortiumsGroupKey].Groups[consortiumName]
+	if !ok {
+		return Organization{}, fmt.Errorf("consortium %s does not exist in channel config", consortiumName)
+	}
+	orgGroup, ok := consortium.Groups[orgName]
+	if !ok {
+		return Organization{}, fmt.Errorf("consortium org %s does not exist in channel config", orgName)
+	}
+
+	org, err := getOrganization(orgGroup, orgName)
+	if err != nil {
+		return Organization{}, err
+	}
+
+	// Remove AnchorPeers which are application org specific.
+	org.AnchorPeers = nil
+
+	return org, err
+}
+
+// newOrgConfigGroup returns an config group for a organization.
+// It defines the crypto material for the organization (its MSP).
+// It sets the mod_policy of all elements to "Admins".
+func newOrgConfigGroup(org Organization) (*cb.ConfigGroup, error) {
+	orgGroup := newConfigGroup()
+	orgGroup.ModPolicy = AdminsPolicyKey
+
+	if err := addPolicies(orgGroup, org.Policies, AdminsPolicyKey); err != nil {
+		return nil, err
+	}
+
+	fabricMSPConfig, err := org.MSP.toProto()
+	if err != nil {
+		return nil, fmt.Errorf("converting fabric msp config to proto: %v", err)
+	}
+
+	conf, err := proto.Marshal(fabricMSPConfig)
+	if err != nil {
+		return nil, fmt.Errorf("marshaling msp config: %v", err)
+	}
+
+	// mspConfig defaults type to FABRIC which implements an X.509 based provider
+	mspConfig := &mb.MSPConfig{
+		Config: conf,
+	}
+
+	err = addValue(orgGroup, mspValue(mspConfig), AdminsPolicyKey)
+	if err != nil {
+		return nil, err
+	}
+
+	// OrdererEndpoints are orderer org specific and are only added when specified for orderer orgs
+	if len(org.OrdererEndpoints) > 0 {
+		err := addValue(orgGroup, endpointsValue(org.OrdererEndpoints), AdminsPolicyKey)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// AnchorPeers are application org specific and are only added when specified for application orgs
+	anchorProtos := []*pb.AnchorPeer{}
```

Comment:
Nit: same as another comment in this file: shouldn't we initialize this slice based on the length of org.AnchorPeers?

Reply:
done